tasks/release_from_changelog.rake

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'bump'
+
+desc 'Generate release notes from current changelog'
+task :release_from_changelog, [:release_type] do |_task, args|
+  def validate_argument(release_type)
+    valid_args = %w[major minor patch pre]
+
+    return if valid_args.include?(release_type)
+    raise ArgumentError,
+          "First argument (release type) must be one of #{valid_args}."
+  end
+
+  def update_readme(old_version, new_version)
+    puts 'Updating the README'
+
+    readme = File.read('README.md')
+
+    File.open('README.md', 'w') do |f|
+      f << readme.sub(
+        "gem 'rubocop', '~> #{old_version}', require: false",
+        "gem 'rubocop', '~> #{new_version}', require: false"
+      )
+    end
+  end
+
+  def update_manual(old_version, new_version)
+    puts 'Updating the manual'
+
+    manual = File.read('manual/installation.md')
+
+    File.open('manual/installation.md', 'w') do |f|
+      f << manual.sub(
+        "gem 'rubocop', '~> #{old_version}', require: false",
+        "gem 'rubocop', '~> #{new_version}', require: false"
+      )
+    end
+  end
+
+  def add_header_to_changelog(version)
+    puts 'Adding header to CHANGELOG'
+
+    changelog = File.read('CHANGELOG.md')
+    head, tail = changelog.split("## master (unreleased)\n\n", 2)
+
+    File.open('CHANGELOG.md', 'w') do |f|
+      f << head
+      f << "## master (unreleased)\n\n"
+      f << "## #{version} (#{Time.now.strftime('%F')})\n\n"
+      f << tail
+    end
+  end
+
+  def create_release_notes(version)
+    puts 'Creating new release notes'
+
+    release_notes = new_version_changes.strip
+    contributor_links = user_links(release_notes)
+
+    File.open("relnotes/v#{version}.md", 'w') do |file|
+      file << release_notes
+      file << "\n\n"
+      file << contributor_links
+    end
+  end
+
+  def new_version_changes
+    changelog = File.read('CHANGELOG.md')
+    _, _, new_changes, _older_changes = changelog.split(/^## .*$/, 4)
+    new_changes
+  end
+
+  def user_links(text)
+    names = text.scan(/\[@(\S+)\]\[\]/).map(&:first).uniq
+    names.map { |name| "[@#{name}]: https://github.com/#{name}" }
+         .join("\n")
+  end
+
+  release_type = args[:release_type]
+  validate_argument(release_type)
+
+  puts 'Bumping version, create git tag and commit'
+
+  old_version = Bump::Bump.current
+  Bump::Bump.run(release_type, tag: true)
```

Comment:
What exactly does this do?

Reply:
It finds & updates the version string, creates a git commit and a git tag.